File: jStuff/ DivisibleSumPairs.java
import java.util.Scanner;

public class Solution {
    /**
    * Returns the number of tuples in ar whose sum divisible by k
    * @param ar - array of size n
    * @param n - the size of array ar
    * @param k - integer which we use to check divisibility
    * @return the number of pairs which are divisibility by k in ar
    */
    static int divisibleSumPairs(int n, int k, int[] ar) {
        int count = 0;
        for (int i = 0; i < ar.length; i++)
            for (int j = i + 1; j < ar.length; j++)
                if( (ar[i] + ar[j]) % k ==0 )
                    count++;
        return count;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int k = in.nextInt();
        int[] ar = new int[n];
        for(int ar_i = 0; ar_i < n; ar_i++){
            ar[ar_i] = in.nextInt();
        }
        int result = divisibleSumPairs(n, k, ar);
        System.out.println(result);
    }
}

File: cppStuff/algorithms/ Divisible_Sum_Pairs.cpp
#include <vector>
#include <iostream>

// pre - vector ar a vector with elements,int n the number of elements in ar, and int k is 
//       passed to the functinon
//post - returns an int count, the number of tuples in ar whose sum is divisible by k
int divisibleSumPairs(int n, int k, std::vector <int> ar) 
{
    int count = 0;
    for ( int i = 0; i < ar.size(); i++ )  
        for ( int j = i + 1; j < ar.size(); j++ ) 
            if ( (ar[i] + ar[j]) % k == 0 )
                count++; 
    return count;
}

int main() 
{
    int n;
    int k;
    std::cin >> n >> k;
    std::vector<int> ar(n);
    for( int ar_i = 0; ar_i < n; ar_i++ )
       std::cin >> ar[ar_i];
    int result = divisibleSumPairs(n, k, ar);
    std::cout << result << '\n';
    return 0;
}